pkg/integrations/v4/emitter/emitter.go

```diff
@@ -98,7 +98,15 @@ func (e *VersionAwareEmitter) Emit(definition integration.Definition, extraLabel
 			elog.WithError(err).WithFields(fields).Warn("can't parse v4 integration output")
 			return err
 		}
-
+		agentResolver := e.aCtx.HostnameResolver()
+		_, overrideHostname, _ := agentResolver.Query()
+
+		for _, dataSet := range pluginDataV4.DataSets {
+			// Only update hostname for windows services
+			if overrideHostname != "" && dataSet.Entity.Type == "WIN_SERVICE" {
+				dataSet.Entity.Metadata["hostname"] = overrideHostname
+			}
```

Comment:
Isn't there any way to do this from the SDK? We should avoid having this integration specific code in the Infra Agent.
If there is no current way to do this in the SDK, maybe we should think in the use case and how could we add it there, so any integration can specify if the hostname should be overridden.

Reply:
I'll look for a way to do this from the SDK,
But `win-services` sends the integration data adds hostname others don't (flex, redis, etc ) so the problem only with win-services (will check other v4 integrations as well), and this part of code only deals with v4 integrations.